src/linux/system.rs

```diff
@@ -541,7 +548,13 @@ impl SystemExt for System {
     }
 
     fn get_os_version(&self) -> Option<String> {
-        get_system_info(InfoType::OsVersion)
+        cfg_if! {
+            if #[cfg(not(target_os = "android"))] {
+                get_system_info(InfoType::OsVersion, Some(BufReader::new(File::open("/etc/os-release").ok()?)))
+            } else {
+                get_system_info::<BufReader<File>>(InfoType::OsVersion, None)
+            }
+        }
```

Comment:
```suggestion
        #[cfg(not(target_os = "android"))]
        {
                get_system_info(InfoType::OsVersion, Some(BufReader::new(File::open("/etc/os-release").ok()?)))
        }
        #[cfg(target_os = "android"))
        {
             BufReader<File>>(InfoType::OsVersion, None)
        }
```

Or equivalent. No need for `cfg_if` in here.

Reply:
Updated